ui/overflow_menu.js

```diff
@@ -50,26 +52,37 @@ shaka.ui.OverflowMenu = class extends shaka.ui.Element {
     this.createChildren_();
 
 
-    const backToOverflowMenuButtons =
-        this.controls.getVideoContainer().getElementsByClassName(
-            'shaka-back-to-overflow-button');
-
-    for (const button of backToOverflowMenuButtons) {
-      this.eventManager.listen(button, 'click', () => {
-        // Hide the submenus, display the overflow menu
-        this.controls.hideSettingsMenus();
-        shaka.ui.Utils.setDisplay(this.overflowMenu_, true);
-
-        // If there are back to overflow menu buttons, there must be
-        // overflow menu buttons, but oh well
-        if (this.overflowMenu_.childNodes.length) {
-          /** @type {!HTMLElement} */ (this.overflowMenu_.childNodes[0])
-              .focus();
-        }
-
-        // Make sure controls are displayed
-        this.controls.computeOpacity();
-      });
+    // Initially, some child buttons inside the overflow menu are created
+    // with a "Close" option.
+    // When inside of the overflow menu, that option must be replaced to
+    // a "Back" arrow that returns the user to the main overflow menu.
+    for (const menu of this.children_) {
+      // Some buttons (like "Loop"), are only toggled on click, without
+      // the need of opening submenus, thus they do not have a backButton
+      // and we must ignore them.
+      const backOrClose = menu.backButton;
+
+      if (backOrClose) {
```

Comment:
Instead of checking for the presence of a `backButton` member variable, maybe you should instead do:
`if (menu instanceof shaka.ui.SettingsMenu) {`

That way, if a user defines a custom UI element that contains a `backButton` member variable, we won't assume that their element is a settings menu.

Reply:
Thank you Theodab!

After researching different approaches, and to provide an extra degree of certainty, I moved the flow to the `SettingsMenu constructor`, this way external components will not be incorrectly identified and the use of the additional conditionals I had incorporated is avoided.